src/Picker/Picker.ios.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import * as React from 'react';
+import { Picker as RNPicker, View } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { StyleSheet } from '../PlatformStyleSheet';
+
+import type { Props } from './PickerTypes';
+
+export type State = {|
+  open: boolean,
+  selectedValue: string,
+  pickerValue: string,
+|};
+
+export default class Picker extends React.Component<Props, State> {
+  static defaultProps = {
+    optionsData: [],
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    const { selectedValue } = this.props;
+
+    this.state = {
+      open: false,
+      selectedValue,
+      pickerValue: selectedValue,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { selectedValue: newSelectedValue } = this.props;
+    const { selectedValue } = this.state;
+    if (
+      newSelectedValue !== prevProps.selectedValue &&
+      newSelectedValue !== selectedValue
+    ) {
+      this.setState({
+        selectedValue: newSelectedValue,
+        pickerValue: newSelectedValue,
+      });
+    }
+  };
+
+  handleOpenPress = () => {
+    this.setState(state => ({
+      open: !state.open,
+    }));
+  };
+
+  handlePickerValueChange = (value: string) => {
+    this.setState({ pickerValue: value });
+  };
+
+  handleOkPress = () => {
+    const { pickerValue } = this.state;
+    const { onValueChange } = this.props;
+    this.setState({
+      open: false,
+      selectedValue: pickerValue,
+    });
+    onValueChange(pickerValue);
+  };
+
+  render() {
+    const { open, selectedValue, pickerValue } = this.state;
+    const { optionsData } = this.props;
+    const selectedOption = optionsData.find(
+      option => option.value === selectedValue
+    );
+    const selectedLabel = selectedOption ? selectedOption.label : '';
+
+    const pickerOptions = optionsData.map(option => (
+      <RNPicker.Item
+        key={option.value}
+        label={option.label}
+        value={option.value}
+      />
+    ));
+
+    return (
+      <View>
+        <Button
+          onPress={this.handleOpenPress}
+          type="secondary"
+          label={selectedLabel}
+          rightIcon={<Icon name="chevron-right" />}
+        />
+        {open && (
+          <View style={styles.container}>
+            <RNPicker
+              selectedValue={pickerValue}
+              style={styles.picker}
+              onValueChange={this.handlePickerValueChange}
+            >
+              {pickerOptions}
+            </RNPicker>
+            <Button onPress={this.handleOkPress} type="secondary" label="OK" />
```

Comment:
I am not sure about this label. I guess OK is pretty international. But we really need to be wary of this. What could we do about it?

Reply:
Label extracted to props as in other cases.